docs: add image directive guide (#47155)
angular/angular: aio/content/guide/image-directive-setup.md

# Setting up `NgOptimizedImage`

> The instructions explain how to setup the `NgOptimizedImage`. For information on using `NgOptimizedImage`, see [Getting Started with NgOptimizedImage](/guide/image-directive).

Reviewer: ```suggestion
This tutorial explains how to setup the `NgOptimizedImage`. For information on using `NgOptimizedImage`, see [Getting Started with NgOptimizedImage](/guide/image-directive).
```

`NgOptimizedImage` provides a generic loader as well as loaders for various third-party image services; it also supports writing your own custom loader.

| Loader type| Behavior |
|:--- |:--- |
| Generic loader | The URL returned by the generic loader will always match the value of `src`. In other words, this loader applies no transformations. Sites that use Angular to serve images are the primary intended use case for this loader.|
| Loaders for third-party image services | The URL returned by the loaders for third-party image services will follow API conventions used by that particular image service. |
| Custom loaders | A custom loader's behavior is defined by its developer. You should use a custom loader if your image service isn't supported by the loaders that come preconfigured with `NgOptimzedImage`.|

Based on the image services commonly used with Angular applications, `NgOptimizedImage` provides loaders preconfigured to work with the following image services:

> | Image Service | Angular API | Documentation |
> |:--- |:--- |:--- |
> | Cloudflare Image Resizing | provideCloudflareLoader | [Documentation](https://developers.cloudflare.com/images/image-resizing/) |
> | Cloudinary | provideCloudinaryLoader | [Documentation](https://developers.cloudflare.com/images/image-resizing/) |  |
> | ImageKit | provideImageKitLoader | [Documentation](https://docs.imagekit.io/) |
> | Imgix | provideImgixLoader | [Documentation](https://docs.imgix.com/) |

Reviewer: ```suggestion
| Image Service | Angular API | Documentation |
|:--- |:--- |:--- |
| Cloudflare Image Resizing | `provideCloudflareLoader` | [Documentation](https://developers.cloudflare.com/images/image-resizing/) |
| Cloudinary | `provideCloudinaryLoader` | [Documentation](https://developers.cloudflare.com/images/image-resizing/) |  |
| ImageKit | `provideImageKitLoader` | [Documentation](https://docs.imagekit.io/) |
| Imgix | `provideImgixLoader` | [Documentation](https://docs.imgix.com/) |
```
nit: code marks for identifiers

> Setting up an image loader is a prequisite to using `NgOptimizedImage`. [These instructions explain how to setup an image loader](/guide/image-loaders.md).

Reviewer: ```suggestion
You must configure an image loader to use `NgOptimizedImage`. [For more details, see the full guide to image loaders](/guide/image-loaders.md).
```
nit: passive voice, rewording second sentence

> The instructions explain how to setup an image loader for use with the `NgOptimizedImage`. 

Reviewer: ```suggestion
These instructions explain how to setup an image loader for use with the `NgOptimizedImage`. 
```

1. Import the `NgOptimizedImage` directive into the application by adding it to the `imports` section of an NgModule or a standalone Component.

<code-example format="typescript" language="typescript">
import { NgOptimizedImage } from '@angular/common';
// Include NgOptimizedImage in the appropriate NgModule
@NgModule({
  imports: [
    // ... other imports
    NgOptimizedImage,
  ],
})

class AppModule {}

> </coode-example>

Reviewer 2: @khempenius the closing tag here got 2 oo here `coode-example` instead of `code-example` - could you please fix?

<code-example format="typescript" language="typescript">
providers: [
  provideImgixLoader('https://my.base.url/'),
],
</code-example>

The base URL for your image assets should be passed to the provider factory as an argument. For most sites, this base URL should match one of the following patterns:

*   https://yoursite.yourcdn.com
*   https://subdomain.yoursite.com
*   https://subdomain.yourcdn.com/yoursite

You can learn more about the base URL structure in the docs of a corresponding CDN provider.

To use a **custom loader**: provide your loader function as a value for the `IMAGE_LOADER` DI token. In the example below, the custom loader function returns a URL starting with `https://example.com` that includes `src` and `width` as URL parameters.

<code-example format="typescript" language="typescript">
providers: [
  {
    provide: IMAGE_LOADER,
    useValue: (config: ImageLoaderConfig) => {
      return `https://example.com/images?src=${config.src}&width=${config.width}`;
    },
  },
],
</code-example>

> A loader function for the `NgOptimizedImage` directive takes an object with the `ImageLoaderConfig` type (the type can be imported from the `@angular/common` package) as its argument and returns the full path of the image asset. The `ImageLoaderConfig` object contains the `src`and `width` properties.

Reviewer: ```suggestion
A loader function for the `NgOptimizedImage` directive takes an object with the `ImageLoaderConfig` type (from `@angular/common`) as its argument and returns the absolute URL of the image asset. The `ImageLoaderConfig` object contains the `src`and `width` properties.
```
* the parenthetical can probably be more concise
* Is "absolute URL" the right term here?
